libkmod/libkmod-index.c

```diff
@@ -233,7 +233,10 @@ static struct index_node_f *index_read(FILE *in, uint32_t offset)
 			return NULL;
 		prefix = strbuf_steal(&buf);
 	} else
-		prefix = NOFAIL(strdup(""));
+		prefix = strdup("");
+
+	if (prefix == NULL)
+		return NULL;
```

Comment:
Unfortunately no. The `strbuf_steal` call might fail due to realloc. Although I doubt that realloc would fail because it would shrink the allocation, the check is at the right position here.